src/parameter.ts

```diff
@@ -29,9 +30,10 @@ export interface Parameter {
   bind?: Binding;
 }
 
-export function assembleParameterSignals(params: Parameter[]) {
+export function assembleParameterSignals(params: (Parameter | TopLevelSelectionDef)[]) {
   const signals: (NewSignal | InitSignal)[] = [];
   for (const param of params || []) {
+    if (isParameterSelection(param)) continue;
```

Comment:
@kanitw, my hunch is that these changes shouldn't be required if I break out a `Generic`/`Normalized` equivalent for `TopLevelProperties` but I'm not entirely sure how to do that and/or what the knock on effect of changes would be. I'd love some help. But I'm also down to leave this as is, because it's a pretty small/localized issue and doesn't even strike me as particularly hacky.

Reply:
I'm going to mark this as resolved for now based on Slack discussion that we will continue working on making it possible to define params/selections at any level of the spec tree.

Reply:
Do we skip parameter selection here because selection has a different codepath? 

If so, this seems fine. But let's add a comment.

Reply:
Yes, separate code path. Comment added with https://github.com/vega/vega-lite/pull/6919/commits/8ddace24bf594503103211f97c5c86762d48d3d3.